src/components/McuUpdateDialogView.tsx

```diff
@@ -93,14 +93,19 @@ const McuUpdateDialogView = () => {
     }, [device, showDelayTimeout]);
 
     const dispatch = useDispatch();
-    const onCancel = () => dispatch(mcubootWritingClose());
+    const onCancel = () => {
+        setKeepDialogOpen(false);
+        dispatch(mcubootWritingClose());
+    };
 
-    const { seconds, minutes, hours, days, isRunning, start, pause, reset } =
-        useStopwatch({ autoStart: true });
+    const { time, start, pause, reset } = useStopwatch({
+        autoStart: false,
+    });
 
     const onWriteStart = useCallback(() => {
         reset();
         start();
+        setKeepDialogOpen(true);
         dispatch(performUpdate(showDelayTimeout ? uploadDelay : null));
     }, [reset, start, dispatch, showDelayTimeout, uploadDelay]);
```

Comment:
I'm not going to block this by being too nitpicky now, but why is `useCallback` used here? `useCallback` and `useMemo` should in practice only be used on computational expensive calls, and this does not look very expensive.
I've heard that if you need to access the same function reference after rerendering, you should rather use `useRef` (https://beta.reactjs.org/reference/react/useRef).